internal/util-interface/src/main/java/xsbti/Problem.java

```diff
@@ -3,10 +3,20 @@
  */
 package xsbti;
 
+import java.util.Optional;
+
 public interface Problem
 {
 	String category();
 	Severity severity();
 	String message();
 	Position position();
-}
\ No newline at end of file
+
+  // Default value to avoid breaking binary compatibility
+  /**
+   * If present, the string shown to the user when displaying this Problem.
+   * Otherwise, the Problem will be shown in an implementation-defined way
+   * based on the values of its other fields.
+   */
+  default Optional<String> rendered() { return Optional.empty(); }
+}
```

Comment:
Can you also add the excellent commit message as documentation here? It explains pretty good why this field is necessary and how it differs from message.

Reply:
Which part of the commit message exactly? I've tried to keep the comment generic enough since the implementation may change.

Reply:
This new addition makes sense, good job.